### PR TITLE
Tn add style

### LINKED ISSUE #10: Add Style
#### As a user I want to see a visually pleasing application.

__Story__:

Our web application is really starting taking shape. We have added a few
features that provide value to our customers, but some of our customers are
complaining that the website is a quite bland. It is time to spice it up by
applying a visually appealing, and easy to maintain user interface to our web
application.

__Acceptance criteria__:
- [x] The application has a header with navigation
- [x] Images are displayed in an aesthetically pleasing list of cards
- [x] Form errors are shown inline with the related input field
- [x] I know how to use Bootstrap components like buttons and navbars in my
  application.
- [x] I know the very basics of CSS and SCSS
- [x] My SCSS files are properly named and organized using BEM convention
- [x] My app has a "sticky" footer.
- [x] My app has a few customized styles using bootstrap variable overrides.
- [x] My app is responsive and is usable on a phone.
- [x] The active link in the navigation header is differentiated.
- [x] I use a bootstrap "container" to center my app contents (not necessarily "fluid").

__Reference material__:
- http://v4-alpha.getbootstrap.com/getting-started/introduction/
- https://github.com/plataformatec/simple_form
- http://getbem.com/naming/

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by justinpearsonappfolio**
Closes #10

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,8 @@ node_modules
 /public/packs-test
 /node_modules
 .idea/
+
+
+# no mac files
+.DS_Store
+
```
Comment (ramonrovirosa) [inline .gitignore]: fix whitespacing. No need for extra spaces on line 29 & 33

File: Gemfile [hunk 2]
```diff
@@ -56,3 +56,6 @@ gem 'acts-as-taggable-on', '~> 5.0'
 
 ## For mock test
 gem 'mocha'
+
+# jpp
```
Comment (ramonrovirosa) [inline Gemfile]: What does this comment mean?

File: app/assets/stylesheets/images.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Example of re-defining a Bootstrap SCSS variable:
```
Comment (ramonrovirosa) [inline app/assets/stylesheets/images.scss]: If we are not using this, would be best to remove. In APM property its best not to include comments like this.

File: app/views/images/_image_to_card.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    <%= image_tag image.url, class: ['card-img','js-image-card-body'] %>
+  </div>
+
+  <div class="card-footer">
+    <% image.tag_list.each do |t| %>
+        <%= link_to t, '#', class: ['card-link js-image-card-tag'] %>
+    <% end %>
+  </div>
+
+</div> <!-- card -->
```
Comment (ramonrovirosa) [inline app/views/images/_image_to_card.html.erb]: Nice, this looks good :) Might be best to not include the comment at the end ```<!-- card -->```

Comment (ramonrovirosa) [reply app/views/images/_image_to_card.html.erb]: Also with BEM we usually use two dashes for Body/Element distinction so e.g. ```class="card--footer"```

File: app/views/images/index.html.erb [hunk 5]
```diff
@@ -1,11 +1,8 @@
-<p>Welcome to Image Sharer!</p>
-<%= link_to "Upload new image URL", new_image_path %>
+<div class="container">
```
Comment (ramonrovirosa) [inline app/views/images/index.html.erb]: might want to also add the class here ```card--container``` so we know this is the container for the card so you would have ```<div class="container card--container">```

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -10,22 +10,51 @@
...
+        <%= image_tag 'logo.png', alt: 'our logo\'s image', style: 'width: 40px;', class: 'some-class' %>
+        <p class="text-muted mx-auto">Best Image Sharer</p>
+      </div>
+      <div class="d-flex">
+        <% [['Home', root_path], ['Upload', new_image_path]].each do |label, path| %>
+            <%= link_to label, path, class: ['nav-item',
+                                             'nav-link',
+                                             'align-self-center',
+                                             "js-#{label.downcase}-button",
+                                             ('active' if current_page?(path))] %>
```
Comment (ramonrovirosa) [inline app/views/layouts/application.html.erb]: Good job, most people dont get this :)

File: config/initializers/simple_form_bootstrap.rb [hunk 7]
```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+# Use this setup block to configure all options available in SimpleForm.
+
+# rubocop:disable Metrics/BlockLength
```
Comment (ramonrovirosa) [inline config/initializers/simple_form_bootstrap.rb]: add these exclusions in the ```.rubocop.yml``` file

File: config/initializers/simple_form_bootstrap.rb [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+# rubocop:disable Metrics/LineLength
+
+SimpleForm.setup do |config|
+  config.error_notification_class = 'alert alert-danger'
```
Comment (ramonrovirosa) [inline config/initializers/simple_form_bootstrap.rb]: Nice, I actually didn't know you could do this. Just learned something

File: app/assets/stylesheets/images.scss [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+//$font-size-base: 2rem;
+
+// Custom bootstrap variables must be set or imported *before* bootstrap.
+@import 'bootstrap';
+
```
Comment (ramonrovirosa) [inline app/assets/stylesheets/images.scss]: Can we add some scss so that when there is a form error, the form input field error message is the color red.

Comment (ramonrovirosa) [reply app/assets/stylesheets/images.scss]: Also can we set a larger width for your input fields on your image form. I want to be able to see the url I inputted.

File: .gitignore (hunk 1 above)
Comment (justinpearsonappfolio) [reply .gitignore]: Done!

File: Gemfile (hunk 2 above)
Comment (justinpearsonappfolio) [reply Gemfile]: It's just my initials. I replaced it with:

```
# We're using Bootstrap 4.0.0 through bundler                                    
# so do not add Bootstrap stylesheet tags or javascript tags                     
# in the main layout.
```

File: app/assets/stylesheets/images.scss (hunk 3 above)
Comment (justinpearsonappfolio) [reply app/assets/stylesheets/images.scss]: Done

File: app/views/images/_image_to_card.html.erb (hunk 4 above)
Comment (justinpearsonappfolio) [reply app/views/images/_image_to_card.html.erb]: Ah -- `card-body` and `card-footer` are Bootstrap built-in names. Should I create some sort of alias to re-name them?

File: app/views/images/index.html.erb (hunk 5 above)
Comment (justinpearsonappfolio) [reply app/views/images/index.html.erb]: Done

File: app/views/layouts/application.html.erb (hunk 6 above)
Comment (justinpearsonappfolio) [reply app/views/layouts/application.html.erb]: Haha, thx. This was the result of a lot of optimizing!

File: config/initializers/simple_form_bootstrap.rb (hunk 8 above)
Comment (justinpearsonappfolio) [reply config/initializers/simple_form_bootstrap.rb]: :)

File: app/views/images/_image_to_card.html.erb (hunk 4 above)
Comment (ramonrovirosa) [reply app/views/images/_image_to_card.html.erb]: In that case its fine, I wouldn't worry about it. I would leave it as card-body & card-footer

File: config/initializers/simple_form_bootstrap.rb (hunk 7 above)
Comment (justinpearsonappfolio) [reply config/initializers/simple_form_bootstrap.rb]: Thanks!

File: app/views/images/_image_to_card.html.erb [hunk 10]
```diff
@@ -10,4 +10,4 @@
     <% end %>
   </div>
```
Comment (ramonrovirosa) [inline app/views/images/_image_to_card.html.erb]: Minor nitpick, but try not to have extra whitespace e.g. line 2, line 6, & line 12 of this file.

File: config/initializers/simple_form_bootstrap.rb [hunk 11]
```diff
@@ -21,7 +21,7 @@
     b.use :label, class: 'control-label'
 
     b.use :input, class: 'form-control'
-    b.use :error, wrap_with: { tag: 'span', class: 'help-block' }
+    b.use :error, wrap_with: { tag: 'span', class: 'help-block text-danger' }
```
Comment (ramonrovirosa) [inline config/initializers/simple_form_bootstrap.rb]: Great, thats what I wanted!

**Comment by justinpearsonappfolio** [comment]
Ok, we just 'amended' the last commit with the whitespace changes you requested!

**Comment by justinpearsonappfolio** [comment]
We found a small bug and fixed it just now, would you plz approve again?

**Pull request merged**